### PR TITLE
feat(lsp): add htmx-lsp

### PR DESCRIPTION
**Pull request by ALameLlama**
## 📑 Description

Adds the HTMX lsp

### CODE REVIEW

File: lua/astrocommunity/lsp/htmx-lsp/init.lua [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    "ThePrimeagen/htmx-lsp",
+    ft = { "htmx" },
+  },
+  {
+    "williamboman/mason-lspconfig.nvim",
+    optional = true,
+    opts = function(_, opts)
+      opts.ensure_installed = require("astrocore").list_insert_unique(opts.ensure_installed, { "htmx" })
+    end,
+  },
```
Comment (Uzaaft) [inline lua/astrocommunity/lsp/htmx-lsp/init.lua:13]:
```suggestion
  {
    "ThePrimeagen/htmx-lsp",
    ft = { "htmx" },
    specs =  {
    	"williamboman/mason-lspconfig.nvim",
	optional = true,
    	opts = function(_, opts)
      		opts.ensure_installed = require("astrocore").list_insert_unique(opts.ensure_installed, { "htmx" })
    	end,
  	},
  },
 
```

Comment (Uzaaft) [reply lua/astrocommunity/lsp/htmx-lsp/init.lua:13]: This might need to be formatted 😂

Comment (mehalter) [reply lua/astrocommunity/lsp/htmx-lsp/init.lua:13]: this is still incorrect.

File: lua/astrocommunity/lsp/htmx-lsp/init.lua [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---@type LazySpec
+return {
+  {
+    "ThePrimeagen/htmx-lsp",
+    ft = { "htmx" },
+  },
```
Comment (mehalter) [inline lua/astrocommunity/lsp/htmx-lsp/init.lua:6]:
```suggestion
```

This is not a neovim plugin.

File: lua/astrocommunity/lsp/htmx-lsp/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# htmx-lsp
+
+LSP for [HTMX](https://htmx.org/
```
Comment (Uzaaft) [inline lua/astrocommunity/lsp/htmx-lsp/README.md:3]:
```suggestion
LSP for [HTMX](https://htmx.org/)
```

File: lua/astrocommunity/lsp/htmx-lsp/README.md
Comment (Uzaaft) [inline lua/astrocommunity/lsp/htmx-lsp/README.md:1]: The readme should note that the htmx-lsp needs to be installed with cargo.

**Pull request closed without merging**